test/DynamicExpresso.UnitTest/GithubIssues.cs

```diff
@@ -221,11 +221,40 @@ static bool GetGFunction2(string arg = null)
 			Assert.Throws<ParseException>(() => interpreter.Eval("GFunction(arg)"));
 
 			// there should be an ambiguous call exception, but GFunction1 is used
-			// because gFunc1.Method.GetParameters()[0].HasDefaultValue == true 
+			// because gFunc1.Method.GetParameters()[0].HasDefaultValue == true
 			// and     gFunc2.Method.GetParameters()[0].HasDefaultValue == false
 			Assert.False((bool)interpreter.Eval("GFunction()"));
 		}
 
+		[Test]
+		public void Github_Issue_159()
+		{
+			// GetGFunction2 is defined inside the test function
+			static bool GetGFunction2(string arg = null)
+			{
+				return arg == null;
+			}
+
+			GFunction gFunc1 = GetGFunction1;
+			GFunction gFunc2 = GetGFunction2;
+
+			var interpreter = new Interpreter();
+			interpreter.SetFunction("GFunction", gFunc1);
+			interpreter.SetFunction("GFunction", gFunc2);
+			interpreter.SetVariable("arg", "arg");
+
+			// ambiguous call
+			var exception = Assert.Throws<ParseException>(() => interpreter.Eval("GFunction(arg)"));
```

Comment:
What happen with standard C# when we have two functions like these?
I would expect to have the same behavior with `SetFunction` and `SetIdentifier` ...

Reply:
There is not an overload relation between `gFunc1` with `gFunc2`, I think it should replace the `gFunc1` delegate when adding the `gFunc2`